pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -265,8 +299,138 @@ func (c *cyclonedxParser) GetPredicates(ctx context.Context) *assembler.IngestPr
 	return preds
 }
 
-func (s *cyclonedxParser) getPackageElement(elementID string) []*model.PkgInputSpec {
-	if packNode, ok := s.packagePackages[string(elementID)]; ok {
+func (c *cyclonedxParser) getVulnerabilities(ctx context.Context, cdxBom *cdx.BOM) error {
+	logger := logging.FromContext(ctx)
+	if cdxBom.Vulnerabilities == nil {
+		logger.Info("no vulnerabilities found in CycloneDX BOM")
+		return nil
+	}
+
+	var status model.VexStatus
+	var justification model.VexJustification
+	var publishedTime time.Time
+	for _, vulnerability := range *c.cdxBom.Vulnerabilities {
+		vuln, err := asmhelpers.CreateVulnInput(vulnerability.ID)
+		if err != nil {
+			return err
+		}
+
+		if vexStatus, ok := vexStatusMap[vulnerability.Analysis.State]; ok {
+			status = vexStatus
+		} else {
+			status = "NOT_PROVIDED"
+		}
+
+		if vexJustification, ok := justificationsMap[vulnerability.Analysis.Justification]; ok {
+			justification = vexJustification
+		} else {
+			justification = model.VexJustificationNotProvided
+		}
+
+		publishedTime, _ = time.Parse(time.RFC3339, vulnerability.Published)
```

Comment:
can default to zero time for this. This way we can be like zero time means it was not specified.